### DISCUSSION TITLE
Sign-posting for first-time contributors: issue comments

### DISCUSSION THREAD

**Discussion by thibaudcolas** [community/community#84205] [Projects and Issues]
### Select Topic Area

Product Feedback

### Feature Area

Issues

### Body

There is currently no specific sign-posting for new contributors arriving on an open issue / pull request. I think that’d help in reducing potentially-unhelpful comments, for example:

- "+1"
- "Please assign me this issue"
- "Can I work on this?"
- "Any news on this?"
- "Here is a summary and proposed solution to this issue according to ChatGPT"

Though those comments aren’t necessarily always problematic, they tend to not bring much to the discussion and still create noise for maintainers. Lots of projects have formal or informal guidelines against such comments.

---

Like existing "first-time contributor" signposting, this could simply be auto-generated with links based on which files are detected in the repository – to give first-time commenters a chance to consult the relevant docs.

This could even be based on the existing sign-posting that’s there for all contributors:

> Remember, contributions to this repository should follow its [contributing guidelines](https://github.com/wagtail/wagtail/blob/eade6be9699346a27e910e1836848900dfc291a1/.github/CONTRIBUTING.md), [security policy](https://github.com/wagtail/wagtail/blob/eade6be9699346a27e910e1836848900dfc291a1/.github/SECURITY.md), and [code of conduct](https://github.com/wagtail/wagtail/blob/eade6be9699346a27e910e1836848900dfc291a1/CODE_OF_CONDUCT.md).

Use the exact same set of links, just make it more visible for first-time contributors. This current sign-posting is below the submit button for the new comment so might as well not be there.

## As-is first-time contributors signposting

For reference, here’s what’s in place currently.

In issues and pull requests listing there’s a lovely banner:

> 👋Want to contribute to wagtail/wagtail?
> 
> If you have a bug or an idea, read the [contributing guidelines](https://github.com/wagtail/wagtail/blob/eade6be9699346a27e910e1836848900dfc291a1/.github/CONTRIBUTING.md) before opening an issue.
> 
> If you're ready to tackle some open issues, [we've collected some good first issues for you](https://github.com/wagtail/wagtail/contribute).

![issues listings](https://github.com/community/community/assets/877585/35db96cd-513c-4fdd-84a8-4e189d39b360)

For new issues / pull requests there’s a dismissable pop-up in the right sidebar over metadata fields:

> 👋
> 
> It looks like this is your first time opening an issue in this project!
> 
> Be sure to review the [contributing guidelines](https://github.com/wagtail/wagtail/blob/eade6be9699346a27e910e1836848900dfc291a1/.github/CONTRIBUTING.md) and [code of conduct](https://github.com/wagtail/wagtail/blob/eade6be9699346a27e910e1836848900dfc291a1/CODE_OF_CONDUCT.md).
> 
> **Reporting a security vulnerability?** Check out the project's [security policy](https://github.com/wagtail/wagtail/security/policy).
> 
> **Looking for help?** Check out the project’s [instructions for getting support](https://github.com/wagtail/wagtail/blob/main/docs/support.md).

![new issue](https://github.com/community/community/assets/877585/f4805db6-64fc-4c59-9a0e-ec0275c1a7f7)

---

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by bryan-shea** [comment]
Currently, there is no specific guidance or "sign-posting" for first-time contributors when they comment on an issue or pull request. This often leads to comments that are well-intentioned but may not add value to the discussion (like "+1", "Can I work on this?", etc.).
- **Impact:** Such comments can create noise and extra work for maintainers, especially in larger projects.

### Proposed Solution:
1. **Enhanced Sign-Posting for First-Time Contributors:**
   - **Auto-Generated Guidance:** Implement an automated system that provides first-time contributors with links to relevant documents (contributing guidelines, code of conduct, etc.) when they attempt to comment on an issue or PR.
   - **Visibility:** Place this guidance in a more prominent position, such as at the top of the comment box or as a pop-up when a first-time contributor starts typing a comment.

2. **Utilize Existing Features with Modifications:**
   - **Adapt Current Sign-Posting:** Use the existing framework for "first-time contributor" notices but make it more visible and context-specific for commenting on issues and PRs.
   - **Positioning:** Ensure that the sign-posting is visible before the contributor types their comment, rather than being placed below the submit button where it's easily missed.

3. **Examples of Enhanced Sign-Posting:**
   - **Issue and PR Listings:** A banner or notice specifically for first-time contributors, offering guidance on how to make constructive comments.
   - **New Issues/PRs Creation:** A sidebar or pop-up for first-time contributors, reminding them to review contribution guidelines and providing links to helpful resources.

### Benefits:
- **Educates New Contributors:** Offers immediate guidance to first-time contributors, helping them understand the norms and expectations of the project.
- **Reduces Noise:** By directing new contributors to relevant guidelines, potentially unhelpful comments are likely to be reduced.
- **Improves Project Management:** Maintainers spend less time managing basic inquiries and more time on substantive contributions.

Implementing a more prominent and context-specific sign-posting system for first-time contributors in issues and pull requests can significantly improve the contribution process for both new contributors and maintainers. This approach aligns with best practices in community management, enhancing the overall health and productivity of open-source projects.

**Reply by thibaudcolas** [reply]
@Bryan-Webifi-dev there’s a lot in there, could you summarize what you’re trying to say? See [BLUF](https://en.wikipedia.org/wiki/BLUF_(communication)) as communication style that works well online.

**Reply by bryan-shea** [reply]
@thibaudcolas 
Apologies for the delay, its been a busy week! Summarized version is below :)

#### BLUF (Bottom Line Up Front): 

- To improve open-source project management, implement a more visible, context-specific sign-posting system for first-time contributors in issues and pull requests. This includes auto-generated guidance with essential document links, more prominent placement of guidance, and adaptation of existing sign-posting features. Benefits include educating new contributors, reducing irrelevant comments, and freeing up maintainers to focus on substantive contributions, ultimately enhancing project health and productivity.

**Reply by stevepiercy** [reply]
Currently I use [GitHub Saved Replies](https://docs.github.com/en/get-started/writing-on-github/working-with-saved-replies/using-saved-replies) to manually manage first-time commenters to our organization and repositories who make irrelevant comments.

We have created, and continue to develop, a separate page in our documentation for [First-time contributors](https://6.docs.plone.org/contributing/first-time.html), and direct newcomers there.

Nonetheless, we would save many volunteer hours per week from an automated tool that stops newcomers from posting irrelevant comments.

**Comment by koppor** [comment]
We use the [assign-issue-action](https://github.com/takanome-dev/assign-issue-action) for working with these kinds of comments.

It detects some magic strings for giving a hint to assign oneself. We do not do automatic assignment on arbitrary texts to route newcomers to our contributing guide. Since the strings are hard coded, we also have `dev: wisdom` label posting the guidance

---

Example output - source: https://github.com/JabRef/jabref/issues/13476#issuecomment-3043511784

![image](https://github.com/user-attachments/assets/a7041574-5e25-4357-a6b3-38772d1a3126)

---

![image](https://github.com/user-attachments/assets/774d9b29-5388-4dc3-9f17-a2685c3635f1)

---

Setup:

- Assign issue on comment: https://github.com/JabRef/jabref/blob/d74e686e02cfb1696227e3218b1b8f4cb7415be8/.github/workflows/on-issue-comment.yml#L15
- Unassign due to inactivity: https://github.com/JabRef/jabref/blob/d74e686e02cfb1696227e3218b1b8f4cb7415be8/.github/workflows/unassign-issues.yml#L17
- Wisdom posting: https://github.com/JabRef/jabref/blob/d74e686e02cfb1696227e3218b1b8f4cb7415be8/.github/workflows/on-issue-labeled.yml#L113
